pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -113,16 +118,67 @@ func (g *GCECloud) GetComputeService() *compute.Service {
 }
 
 type rateLimitedRoundTripper struct {
-	rt      http.RoundTripper
-	limiter flowcontrol.RateLimiter
+	rt          http.RoundTripper
+	getLimiter  flowcontrol.RateLimiter
+	postLimiter flowcontrol.RateLimiter
 }
 
 func (rl *rateLimitedRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
-	startTime := time.Now()
-	rl.limiter.Accept()
-	// TODO: Reduce verbosity once #26119 is fixed.
-	glog.V(0).Infof("GCE api call: %s %s (throttled for %v)", req.Method, req.URL.String(), time.Now().Sub(startTime))
-	return rl.rt.RoundTrip(req)
+	for tries := 1; ; tries++ {
+		startTime := time.Now()
+		// Split up the GET and POST methods and a poor man's priority
+		// queue between updates and operation polling. (This is
+		// lopsided at startup when we slam GCE with route create
+		// requests, but after that the POST requests will get rate
+		// limited much less.)
+		if req.Method == http.MethodGet || req.Method == http.MethodHead {
+			rl.getLimiter.Accept()
+		} else {
+			rl.postLimiter.Accept()
+		}
+		// TODO: Reduce verbosity once #26119 is fixed.
+		glog.V(0).Infof("GCE api call: %s %s (throttled for %v)", req.Method, req.URL.String(), time.Now().Sub(startTime))
+		res, err := rl.rt.RoundTrip(req)
+		if err != nil || res.StatusCode != http.StatusForbidden {
+			return res, err
+		}
+
+		// Okay, we know it's a 403 of some sort. Now let's try fo
+		// figure out of it's a Rate Limit Exceeded. Get the body,
+		// then reset it, because we're just an interposing
+		// RoundTripper, not the final hop.
+		slurp, err := ioutil.ReadAll(res.Body)
```

Comment:
I'm not 100% convinced we want to retry at this layar, but I'm not going to block this.

However, please encapsulate this code in a function (isRateLimitExceeded or sth like that).